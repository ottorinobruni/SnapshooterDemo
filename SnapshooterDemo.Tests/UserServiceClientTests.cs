using Newtonsoft.Json;
using Snapshooter.Xunit;

public class UserServiceClientTests
{
    [Fact]
    public void CreateUser_ShouldReturnValidUserJson()
    {
        // Arrange
        var serviceClient = new UserServiceClient();
        var userId = Guid.Parse("1292F21C-8501-4771-A070-C79C7C7EF451");
        var firstName = "Otto";
        var lastName = "Bruni";
        var age = 40;
        var email = "ottorino.bruni@gmail.com";

        // Act
        var resultJson = serviceClient.CreateUser(userId, firstName, lastName, email, age);

        // Assert
        //Snapshot.Match(resultJson);
        Snapshot.Match(resultJson, matchOptions => matchOptions.IgnoreField("RegistrationDate"));
    }

    [Fact]
    public void CreateUserWithoutSnapshot_ShouldReturnValidUserJson()
    {
        // Arrange
        var serviceClient = new UserServiceClient();
        var userId = Guid.Parse("1292F21C-8501-4771-A070-C79C7C7EF451");
        var firstName = "Otto";
        var lastName = "Bruni";
        var age = 40;
        var email = "ottorino.bruni@gmail.com";

        // Act
        var resultJson = serviceClient.CreateUser(userId, firstName, lastName, email, age);
        var deserializedUser = JsonConvert.DeserializeObject<User>(resultJson);

        // Assert
        Assert.NotNull(deserializedUser);
        Assert.Equal(userId, deserializedUser.UserId);
        Assert.Equal(firstName, deserializedUser.FirstName);
        Assert.Equal(lastName, deserializedUser.LastName);
        Assert.Equal(email, deserializedUser.Email);
        Assert.Equal(age, deserializedUser.Age);
    }
}