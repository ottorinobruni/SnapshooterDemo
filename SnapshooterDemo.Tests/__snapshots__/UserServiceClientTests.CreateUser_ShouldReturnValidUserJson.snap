{"UserId":"1292f21c-8501-4771-a070-c79c7c7ef451","FirstName":"Otto","LastName":"Bruni","Age":40,"Email":"ottorino.bruni@gmail.com","RegistrationDate":"2024-03-08T14:45:12.7666296Z"}
